File: 414.third-maximum-number.ts
/*
 * @lc app=leetcode id=414 lang=typescript
 *
 * [414] Third Maximum Number
 */

// @lc code=start
function thirdMax(nums: number[]): number {
		const uniqueNums = Array.from(new Set(nums));
		uniqueNums.sort((a, b) => b - a);
		
		return uniqueNums.length >= 3 ? uniqueNums[2] : uniqueNums[0];
};
// @lc code=end

File: 414.third-maximum-number.py
#
# @lc app=leetcode id=414 lang=python3
#
# [414] Third Maximum Number
#

# @lc code=start
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = sorted(set(nums), reverse=True)
        return nums[2] if len(nums) >= 3 else nums[0]
# @lc code=end

